packages/slate-react/src/components/editable.tsx

```diff
@@ -539,13 +539,24 @@ export const Editable = (props: EditableProps) => {
           },
           [attributes.onCompositionEnd]
         )}
+        onCompositionUpdate={useCallback(
+          (event: React.CompositionEvent<HTMLDivElement>) => {
+            if (
+              hasEditableTarget(editor, event.target) &&
+              !isEventHandled(event, attributes.onCompositionUpdate)
+            ) {
+              state.isComposing = true
+            }
+          },
+          [attributes.onCompositionUpdate]
+        )}
         onCompositionStart={useCallback(
           (event: React.CompositionEvent<HTMLDivElement>) => {
             if (
               hasEditableTarget(editor, event.target) &&
               !isEventHandled(event, attributes.onCompositionStart)
             ) {
-              state.isComposing = true
+              editor.exec({ type: 'delete_fragment' })
```

Comment:
Because cursor position needs to be in sync after DOM nodes are removed by executing the following lines.
https://github.com/ianstormtaylor/slate/blob/a55396230cb2c20a614777c94623e401b7860d17/packages/slate-react/src/components/editable.tsx#L181-L200

Otherwise cursor doesn't work properly, like this:

![ezgif com-crop](https://user-images.githubusercontent.com/29003390/71160041-2cc28c00-228a-11ea-8a87-f172e9d370e2.gif)

Reply:
when you typing あい was at wrong place

Reply:
The above gif shows wrongly fixed version (i.e. `state.isComposing = true` is in composition**start** event not in composition**update** event).